### PR TITLE
Nasm to rust

### PR DESCRIPTION
**Pull request by lu-zero**
Since @barrbrain wanted an example. It is less straightforward than it should indeed.

### CODE REVIEW

**Comment by lu-zero** [comment]
There is something really fishy going on. Reproduce the dav1d failures locally.

File: src/asm/x86/mc.rs [hunk 1]
```diff
@@ -363,3 +363,88 @@ pub(crate) static AVG_FNS: [Option<AvgFn>; CpuFeatureLevel::len()] = {
 
 pub(crate) static AVG_HBD_FNS: [Option<AvgHBDFn>; CpuFeatureLevel::len()] =
   [None; CpuFeatureLevel::len()];
+
+#[rustfmt::skip]
+static MC_SUBPEL_FILTERS: AlignedArray<[i8; 600]> = AlignedArray::new([
```
Comment (barrbrain) [inline src/asm/x86/mc.rs]: The following change passes tests for me with feature `decode_test_dav1d`:
```suggestion
#[no_mangle]
pub static rav1e_mc_subpel_filters: AlignedArray<[i8; 600]> = AlignedArray::new([
```

File: src/asm/x86/mc.rs [hunk 2]
```diff
@@ -363,3 +363,88 @@ pub(crate) static AVG_FNS: [Option<AvgFn>; CpuFeatureLevel::len()] = {
...
+0,   0,   6,  26,  26,   6,   0,   0,
+0,   0,   6,  24,  27,   7,   0,   0,
+0,   0,   5,  23,  28,   8,   0,   0,
+0,   0,   4,  22,  29,   9,   0,   0,
+0,   0,   3,  21,  30,  10,   0,   0,
+0,   0,   2,  20,  31,  11,   0,   0,
+0,   0,   2,  18,  31,  13,   0,   0,
+0,   0,   1,  17,  31,  15,   0,   0 ]);
+
+#[no_mangle]
```
Comment (barrbrain) [inline src/asm/x86/mc.rs]:
```suggestion
```

File: src/asm/x86/mc.rs [hunk 3]
```diff
@@ -363,3 +363,88 @@ pub(crate) static AVG_FNS: [Option<AvgFn>; CpuFeatureLevel::len()] = {
...
+pub static rav1e_mc_subpel_filters: &[i8] = &MC_SUBPEL_FILTERS.array;
```
Comment (barrbrain) [inline src/asm/x86/mc.rs]:
```suggestion
```

File: src/asm/x86/mc.rs (hunk 1 above)
Comment (lu-zero) [reply src/asm/x86/mc.rs]: Right, that was my doubt, I still do not understand how it could work for the aom tests though.

**Comment by barrbrain** [comment]
Thanks for the insight. This was handled in #1763.

**Pull request closed without merging**